test/test_tensor.py

```diff
@@ -239,10 +239,14 @@ def test_tensor_ndarray_dtype(self):
     assert Tensor(arr, dtype=dtypes.float64).dtype == dtypes.float64 # check that it works for something else
 
   def test_tensor_list_dtype(self):
-    arr = [1]
-    assert Tensor(arr).dtype == dtypes.int32
-    assert Tensor(arr, dtype=dtypes.float32).dtype == dtypes.float32
-    assert Tensor(arr, dtype=dtypes.float64).dtype == dtypes.float64
+    for arr in ([1], [[1,1],[1,1]], [[[1,1],[1,1]],[[1,1],[1,1]]]):
+      assert Tensor(arr).dtype == dtypes.int32
+      assert Tensor(arr, dtype=dtypes.float32).dtype == dtypes.float32
+      assert Tensor(arr, dtype=dtypes.float64).dtype == dtypes.float64
+    assert Tensor([[0,1],[3.,4]]).dtype == Tensor.default_type
+    assert Tensor([[[0],[1]],[[3.],[4]]]).dtype == Tensor.default_type
```

Comment:
also test the shape and content with this, not just the dtypes. does `[[[]]]` work? try to think about unique cases that enabled by this feature, and how it can fail, and improve the coverage while you are at it

Reply:
oooooo yeah you're right.
I should've thought about all the cases.

Reply:
```
In [17]: all_int([])
Out[17]: True
```
caught this too

Thanks for the guidance :D